packages/clay-navigation-bar/src/index.tsx

```diff
@@ -0,0 +1,151 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React from 'react';
+
+import ClayIcon from '@clayui/icon';
+import ClayLink from '@clayui/link';
+import warning from 'warning';
+
+import {useTransition} from './hooks';
+
+interface ItemProps extends React.HTMLAttributes<HTMLLIElement> {
+	/**
+	 * Determines the active state of an dropdown list item.
+	 */
+	active?: boolean;
+
+	/**
+	 * Children elements received from ClayNavigationBar.Item component.
+	 */
+	children: React.ReactElement;
+}
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * Children elements received from ClayNavigationBar component.
+	 */
+	children: React.ReactElement<ItemProps>[];
+
+	/**
+	 * Determines the style of the Navigation Bar
+	 */
+	inverted?: boolean;
+
+	/**
+	 * Path to the location of the spritemap resource.
+	 */
+	spritemap: string;
+
+	/**
+	 * Set up dropdown's trigger label.
+	 */
+	triggerLabel: string;
+}
+
+type ItemType = React.FunctionComponent<ItemProps>;
+
+const Item: ItemType = ({active, children, className, ...otherProps}) => {
+	return (
+		<li {...otherProps} className={classNames('nav-item', className)}>
+			{React.Children.map(
+				children,
+				(child: React.ReactElement<ItemProps>, index) =>
+					React.cloneElement(child, {
+						...child.props,
+						className: classNames(child.props.className, {
+							active,
+						}),
+						key: index,
+					})
+			)}
+		</li>
+	);
+};
+
+const ClayNavigationBar: React.FunctionComponent<Props> & {
+	Item: ItemType;
+} = ({
+	children,
+	className,
+	inverted = false,
+	spritemap,
+	triggerLabel,
+	...otherProps
+}) => {
+	const [visible, setVisible] = React.useState(false);
+	const contentRef = React.useRef<HTMLDivElement>(null);
+	const [
+		transitioning,
+		handleTransitionEnd,
+		handleClickToggler,
+	] = useTransition(visible, setVisible, contentRef);
+
+	const activeElementsCount = children.filter(child => child.props.active)
+		.length;
+
+	warning(
+		activeElementsCount <= 1,
+		`ClayNavigationBar expects 0 or 1 active children, but received ${activeElementsCount}`
+	);
+
+	return (
+		<nav
+			{...otherProps}
+			className={classNames(
+				className,
+				'navbar',
+				'navbar-collapse-absolute',
+				'navbar-expand-md',
+				'navbar-underline',
+				'navigation-bar',
+				{
+					'navigation-bar-light': !inverted,
+					'navigation-bar-secondary': inverted,
+				}
+			)}
+		>
+			<div className="container-fluid container-fluid-max-xl">
+				<ClayLink
```

Comment:
hey @diegonvs, this should probably be a `ClayButton` or `button`, since we are not passing `href` and we are controlling your click. What do you think?

Reply:
Using ClayButton the element looks more dimmed. I consider ClayLink more suitable for the trigger, even element isn't the more semantic for it.

What do you think @pat270?

With ClayButton, tested with all displayTypes:
<img width="443" alt="Screen Shot 2019-05-13 at 14 41 01" src="https://user-images.githubusercontent.com/7663513/57642245-34e87980-758d-11e9-804f-276634c5d25e.png">

With ClayLink
<img width="447" alt="Screen Shot 2019-05-13 at 14 41 13" src="https://user-images.githubusercontent.com/7663513/57642249-37e36a00-758d-11e9-9c6e-2a990b74a878.png">

Reply:
@diegonvs You can use `<button class="btn btn-unstyled nav-link">` as long as ClayButton supports it. Those classes should make it look the same as the link.